application/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/mapper/FaCatchMapper.java

```diff
@@ -196,16 +190,31 @@ protected String getSizeDistributionClassCodeListId(SizeDistribution sizeDistrib
         return fluxCharacteristicEntities;
     }
 
-    protected Set<FluxLocationEntity> getFluxLocationEntities(List<FLUXLocation> fluxLocations) {
-        Set<FluxLocationEntity> fluxLocationEntities = new HashSet<>();
+    protected Set<LocationEntity> getFluxLocationEntities(List<FLUXLocation> fluxLocations, FaCatchEntity faCatchEntity) {
+        if (fluxLocations == null || fluxLocations.isEmpty()) {
+            return Collections.emptySet();
+        }
         FluxLocationDao fluxLocationDao = new FluxLocationDao(em);
+        Set<LocationEntity> fluxLocationEntities = new HashSet<>();
+        // TODO: POSITIONS!!!!
         for (FLUXLocation fluxLocation : Utils.safeIterable(fluxLocations)) {
-            FluxLocationEntity fluxLocationEntity = fluxLocationDao.findLocation(fluxLocation.getID());
-            if(fluxLocationEntity == null) {
-                fluxLocationEntity = locationMapper.mapToFluxLocationEntity(fluxLocation);
-                em.persist(fluxLocationEntity);
+            if(fluxLocation.getTypeCode() != null && fluxLocation.getTypeCode().getValue().equals("POSITION")) {
+                FLUXGeographicalCoordinate coordinate = fluxLocation.getSpecifiedPhysicalFLUXGeographicalCoordinate();
+                if(coordinate != null) {
+                    faCatchEntity.setLongitude(coordinate.getLongitudeMeasure().getValue().doubleValue());
```

Comment:
Maybe split this method into two because there is no hint that this method updates the catch entity as well as return a list of locations

Reply:
done